### PR TITLE
Miscellaneous JavaScript cleanup

### PR DESCRIPTION
**Pull request by tsa96**
Now 0.9.1 is basically done I wanted to go through all our old JS and improve the quality of the code.

Also includes a quick fix for https://github.com/momentum-mod/game/issues/1755

### CODE REVIEW

File: scripts/mainmenu.js [hunk 1]
```diff
@@ -1,115 +1,147 @@
...
-			$.Msg(
-				$.persistentStorage.key(i).toString() +
-					': ' +
-					JSON.stringify($.persistentStorage.getItem($.persistentStorage.key(i)))
-			);
-		}
-	}
+	/**
+	 * General onLoad initialisations.
+	 * Fired when ChaosMainMenu fires it's onload event.
```
Comment (Gocnak) [inline scripts/mainmenu.js:55]: its*

File: scripts/mainmenu.js [hunk 2]
```diff
@@ -1,115 +1,147 @@
...
+		this.showPlaytestWelcomePopup();
 	}
 
+	/**
+	 * Fired by C++ whenever main menu is switched to.
+	 */
 	static onShowMainMenu() {
-		MainMenuController.onInitFadeUp();
+		this.panels.movie = $('#MainMenuMovie');
+		this.panels.image = $('#MainMenuBackground');
```
Comment (Gocnak) [inline scripts/mainmenu.js:89]: Why's this happening here again? Static initialization sets this already.

File: scripts/mainmenu.js [hunk 3]
```diff
@@ -1,115 +1,147 @@
...
 		// Check to see if tab to show exists.
 		// If not load the xml file.
-		if (!$.GetContextPanel().FindChildInLayoutFile(tab)) {
-			const newPanel = $.CreatePanel('Panel', this.contentPanel, tab);
+		if (!this.panels.cp.FindChildInLayoutFile(tab)) {
+			const newPanel = $.CreatePanel('Panel', this.panels.pageContent, tab);
 
-			newPanel.Data().elMainMenuRoot = $.GetContextPanel();
-			newPanel.LoadLayout('file://{resources}/layout/' + xmlName + '.xml', false, false);
+			// newPanel.Data().elMainMenuRoot = $.GetContextPanel();
```
Comment (Gocnak) [inline scripts/mainmenu.js:138]: Commented out code?

File: scripts/mainmenu_mapselector.js [hunk 4]
```diff
@@ -513,16 +515,14 @@ class MapSelection {
...
-		}
+		for (let type of Array(3).keys()) panel.SetHasClass(MapSelNStateClasses[type], state === type);
 	}
 
 	/**
 	 * Only show the clear button if the search is not empty
 	 */
 	static onSearchChanged() {
-		MapSelection.searchClearButton.SetHasClass('search__clearbutton--hidden', this.searchTextEntry.text === '');
+		this.panels.searchText.SetHasClass('search__clearbutton--hidden', this.searchText.text === '');
```
Comment (Gocnak) [inline scripts/mainmenu_mapselector.js:525]: The second `this.searchText` needs to be `this.panels.searchText`?

File: scripts/mainmenu_settings.js [hunk 5]
```diff
@@ -407,40 +417,40 @@ class MainMenuSettings {
...
+			if (this.panels.info.HasClass('settings-info--hidden')) {
+				this.panels.info.RemoveClass('settings-info--hidden');
 			} else {
 				switchDelay = 0.05; // This should always be half the duration of settings-info--switch
 
-				this.infoPanel.AddClass('settings-info--switch');
-				let kfs = this.infoPanel.CreateCopyOfCSSKeyframes('BlurFadeInOut');
-				this.infoPanel.UpdateCurrentAnimationKeyframes(kfs);
+				this.panels.info.AddClass('settings-info--switch');
+				let kfs = this.panels.info.CreateCopyOfCSSKeyframes('BlurFadeInOut');
```
Comment (Gocnak) [inline scripts/mainmenu_settings.js:426]: `kfs` can be const

**Review by Gocnak — CHANGES_REQUESTED** [review]

File: scripts/mainmenu.js (hunk 2 above)
Comment (tsa96) [reply scripts/mainmenu.js:89]: I don't understand why, but loading into a map unsets these, resulting in a black background after you disconnect back to main menu. Easiest fix is to just set them again and call `setMainMenuBackground` again.

File: scripts/mainmenu.js (hunk 3 above)
Comment (tsa96) [reply scripts/mainmenu.js]: Oh yeah, think this is from CS:GO and I forgot to remove it. Interesting what they're doing here, passing a parent panel's JSO into a child panel.

File: scripts/mainmenu_mapselector.js (hunk 4 above)
Comment (tsa96) [reply scripts/mainmenu_mapselector.js]: Good catch

File: scripts/mainmenu_settings.js (hunk 5 above)
Comment (tsa96) [reply scripts/mainmenu_settings.js]: Donion rings

File: scripts/mainmenu.js (hunk 1 above)
Comment (tsa96) [reply scripts/mainmenu.js]: My bad, forgot this one, just pushed

**Review by bltech — APPROVED** [review]

File: scripts/mainmenu_mapselector.js [hunk 6]
```diff
@@ -202,7 +204,7 @@ class MapSelection {
 	 * Clear the search bar.
 	 */
 	static clearSearch() {
-		this.searchTextEntry.text = '';
+		this.searchText.text = '';
```
Comment (Gocnak) [inline scripts/mainmenu_mapselector.js:207]:
```suggestion
		this.panels.searchText.text = '';
```

File: scripts/mainmenu_mapselector.js [hunk 7]
```diff
@@ -213,25 +215,27 @@ class MapSelection {
...
-			button.currentstate = 0;
-		});
+		[
+			this.panels.completedFilterButton,
+			this.panels.favoritesFilterButton,
+			this.panels.downloadedFilterButton
+		].forEach((button) => (button.currentstate = 0));
 
 		// Reset tier slider
 		this.tierSlider.SetValues(TIER_MIN, TIER_MAX);
```
Comment (Gocnak) [inline scripts/mainmenu_mapselector.js:241]:
```suggestion
		this.panels.tierSlider.SetValues(TIER_MIN, TIER_MAX);
```

Comment (Gocnak) [reply scripts/mainmenu_mapselector.js:241]: Honestly go through this file and triple check and make sure all the panels are properly referenced by the `this.panels.` notation now.

**Review by Gocnak — CHANGES_REQUESTED** [review]

File: scripts/mainmenu_mapselector.js (hunk 7 above)
Comment (tsa96) [reply scripts/mainmenu_mapselector.js]: Yikes, sorry I did such a crap job on this. Been through everything thoroughly this time.

**Review by Gocnak — APPROVED** [review]

**Pull request merged**